### PR TITLE
Deposit long Otoken

### PR DESCRIPTION
**Pull request by haythem96**
# Task: Feature Name:
Controller module: Deposit long Otoken

## High Level Description

This PR add the functionality of depositing a long Otoken into a vault.
Should be reviewed/merged after #92 .

### Code

- [ ] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by antoncoding** [comment]
One un-related feedback:
We should use rebase more... the git history is getting messy because it contain all the changes from the controller skeleton PR.

File: contracts/Controller.sol [hunk 1]
```diff
@@ -310,7 +322,30 @@ contract Controller is ReentrancyGuard, Ownable {
      * @notice deposit long option into vault
      * @param _args DepositArgs structure
      */
-    // function _depositLong(Actions.DepositArgs memory _args) internal {}
+    function _depositLong(Actions.DepositArgs memory _args) internal returns (MarginAccount.Vault memory) {
+        require(_args.from == msg.sender, "Controller: depositor address and msg.sender address mismatch");
+
+        address whitelistModule = AddressBookInterface(addressBook).getWhitelist();
+        WhitelistInterface whitelist = WhitelistInterface(whitelistModule);
```
Comment (antoncoding) [inline contracts/Controller.sol:329]: You can probably have an internal function called `getWhitelist()` that returns a WhitelistInterface, also the same for MarginPoolInterface, I can see we will use this a lot in the Controller. 

Will only save you 1 line of code but I think it helps readability a bit,  the naming of `whitelistModule` and `whitelist` is not that helpful 😂

Comment (haythem96) [reply contracts/Controller.sol:329]: Yeah I agree, I'm gonna refactor this to something more fancy than that in another PR 😋

File: contracts/Controller.sol [hunk 2]
```diff
@@ -310,7 +322,30 @@ contract Controller is ReentrancyGuard, Ownable {
      * @notice deposit long option into vault
      * @param _args DepositArgs structure
      */
-    // function _depositLong(Actions.DepositArgs memory _args) internal {}
+    function _depositLong(Actions.DepositArgs memory _args) internal returns (MarginAccount.Vault memory) {
+        require(_args.from == msg.sender, "Controller: depositor address and msg.sender address mismatch");
```
Comment (antoncoding) [inline contracts/Controller.sol:326]: wasn't the conclusion for args.from also be acceptable to be the vault owner?
Let's say msg.sender is an operator, the operator can also specify the `owner` to be the args.from. In that case our marginPool will just try to pull assets from the vault owner.

**Review by antoncoding — APPROVED** [review]
Look good! The tests are pretty clean especially the part for multiple actions.
I think the args.sender discussion worth a whole other thread, no need to resolve for this PR 🔥

File: contracts/Controller.sol (hunk 2 above)
Comment (aparnakr) [reply contracts/Controller.sol:296]: for later down the line, not right now.

Comment (aparnakr) [reply contracts/Controller.sol:296]: not in this version, we said its too dangerous to open that up.

**Pull request merged**